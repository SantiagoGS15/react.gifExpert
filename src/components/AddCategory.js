import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
  
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue?.trim().length > 2){
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    return (
    <form onSubmit={handleSubmit}>
        <h1>Add Category</h1>
        <input 
            value={inputValue}
            onChange={handleInputChange}

            type= "text"
        />
    </form>);
};


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};