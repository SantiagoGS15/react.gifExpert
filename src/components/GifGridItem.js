import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
  return (<div className="card">
      <img src={url} alt={title}></img>
      <p className="animate__animated animate__bounce">{title}</p>
  </div>);
};


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};