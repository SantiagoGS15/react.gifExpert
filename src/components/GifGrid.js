import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);
  
    return (
      <>
        <h3 className="animate__fadeIn">{category}</h3>
        {loading && <p>Loading...</p>}
        <div className="card-grid">
          
          <ol>
            {
              images.map((item) => {
                return <GifGridItem 
                      {...item}
                      key={item.id}>
                    </GifGridItem>
              })
            }
          </ol>
        </div>
      </>
  );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
