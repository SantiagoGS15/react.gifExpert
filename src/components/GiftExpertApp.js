import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';
import PropTypes from 'prop-types';

export const GiftExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);
    
    return (<>
        <h2>GiftExpertApp</h2>
        <AddCategory setCategories={setCategories}></AddCategory>
        <hr></hr>
        
        <ol>
            {
               categories.map((category) => {
                   return <GifGrid category={category} key={category}/>
               })
            }
        </ol>
    </>);
};

GiftExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}