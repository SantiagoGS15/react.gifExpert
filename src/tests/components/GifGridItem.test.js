import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import React from 'react'

describe('Pruebas en GifGridItem', () => {

    const title = 'Test';
    const url = 'http://localhost:5000/test.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
    test('debe de mostrar el tÃ­tulo correctamente', () => {
        expect(wrapper.find('p').text().trim()).toBe(title);
    });
});