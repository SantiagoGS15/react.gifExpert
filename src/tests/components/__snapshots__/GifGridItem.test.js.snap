// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="card"
>
  <img
    alt="Test"
    src="http://localhost:5000/test.jpg"
  />
  <p
    className="animate__animated animate__bounce"
  >
    Test
  </p>
</div>
`;
